pkg/operator/ceph/cluster/cluster.go

```diff
@@ -217,6 +218,10 @@ func (c *ClusterController) initializeCluster(cluster *cluster) error {
 
 	// Start the monitoring if not already started
 	c.configureCephMonitoring(cluster, cluster.ClusterInfo)
+
+	// Asynchronously report the telemetry to allow another reconcile to proceed if needed
+	go cluster.reportTelemetry()
+
```

Comment:
I'm a bit concerned what might happen if Ceph is hanging and causing multiple reconciles in fairly quick succession while also leaving us with a lot of these telemetry goroutines running at the same time trying to update telemetry details. I think it's best if we run the reportTelemetry() function without a new goroutine, OR we can use a `sync.Mutex` to make sure we aren't running more than one `reportTelemetry()` routine. The first is simpler, but the latter shouldn't increase our reconcile time much.

Reply:
Agreed, it's a good exercise to consider what would happen in the worst case...

This is at the end of a reconcile. The telemetry will only be reported after a successful reconcile. If there was any error that caused the reconcile to fail, the reconcile would have already aborted. The likelihood of ceph failing at this point should be very small since the reconcile just succeeded. 

If ceph did become unresponsive during the telemetry reporting and another reconcile happened to kick off, both telemetry and the new reconcile would get stuck. Then once ceph begins responding again, both the reconcile and the telemetry should be able to complete. The reconcile typically has many ceph commands and would take much longer than the telemetry goroutine to complete, so I don't see how more than one telemetry goroutine could realistically get backed up.

In the case that multiple goroutines did end up starting, the side effects seem really minimal. They are just ceph commands that would be waiting for ceph to respond. 

If we do find a likely failure case with negative side effects, agreed that we should add safeguards as suggested. Can you expand on where you see how multiple reconciles could happen in quick succession where telemetry would get backed up even while reconciles are completing?

Reply:
Thinking about this a bit more, implementing the mutex is so simple and removes all concern for this discussion, so i'll just go ahead with that change.

Reply:
Ah, sounds good. I missed that it was at the end. But since it's easy and here, I guess it's still safer "just in case"